### DISCUSSION TITLE
Use-case scenarios for MapperIgnoreTargetAttribute and MapperIgnoreSourceAttribute

### DISCUSSION THREAD

**Discussion by ristogod** [riok/mapperly#875] [Q&A]
I have a simple question around how to correctly use the two competing types of MapperIgnore attributes.

When we use just MapperIgnoreTargetAttribute, we get warnings of RMG020.

When we use just MapperIgnoreSourceAttribute, we get warnings of RMG012.

So if we have a method like so (this example is using identical type for source and target):

``` csharp
   [MapperIgnoreTarget(nameof(Models.Image.AccountId))]
   [MapperIgnoreTarget(nameof(Models.Image.CreatedTs))]
   [MapperIgnoreTarget(nameof(Models.Image.Id))]
   [MapperIgnoreTarget(nameof(Models.Image.LastUpdatedTs))]
   [MapperIgnoreTarget(nameof(Models.Image.ParentId))]
   public partial Models.Image Copy(Models.Image source);
```

Do we have to use both (like below) to ensure no warnings appear? 

``` csharp
   [MapperIgnoreTarget(nameof(Models.Image.AccountId))]
   [MapperIgnoreTarget(nameof(Models.Image.CreatedTs))]
   [MapperIgnoreTarget(nameof(Models.Image.Id))]
   [MapperIgnoreTarget(nameof(Models.Image.LastUpdatedTs))]
   [MapperIgnoreTarget(nameof(Models.Image.ParentId))]
   [MapperIgnoreSource(nameof(Models.Image.AccountId))]
   [MapperIgnoreSource(nameof(Models.Image.CreatedTs))]
   [MapperIgnoreSource(nameof(Models.Image.Id))]
   [MapperIgnoreSource(nameof(Models.Image.LastUpdatedTs))]
   [MapperIgnoreSource(nameof(Models.Image.ParentId))]
   public partial Models.Image Copy(Models.Image source);
```

The generated code seem to the same regardless of whether we use or the other or both.

**Comment by TimothyMakkison** [comment]
Hey, you'll need to ignore both the source and target members.
RMG012 and RMG020 are warnings to tell the user that some members are not mapped. By ignoring a member, Mapperly will not attempt to map it or generate the warning.

In a future release you'll be able to tell Mapperly to not warn you about unmapped members.
